core/commands/add.go

```diff
@@ -218,6 +220,101 @@ You can now check what blocks have been created by:
 
 		opts = append(opts, nil) // events option placeholder
 
+		revival_target_cid, _ := req.Options[reviveOptionName].(string)
+
+		var default_options = []cmds.OptMap{
+			cmds.OptMap{
+				"inline-limit": 32,
+				"chunker":      "size-262144",
+				"quiet":        true,
+				"hash":         "sha2-512",
+				"encoding":     "json",
+			},
+			cmds.OptMap{
+				"inline-limit": 32,
+				"chunker":      "size-262144",
+				"quiet":        true,
+				"hash":         "sha2-256",
+				"encoding":     "json",
+			},
+		}
+
+		if revival_target_cid != "" {
+
+			for index, default_opts := range default_options {
+				special := req.Files
+				fmt.Println("\n - - - - - ", index, "\n")
+				fmt.Println(default_opts)
+
+				inlineLimit, _ := default_opts[inlineLimitOptionName].(int)
+				chunker, _ := default_opts[chunkerOptionName].(string)
+				hashFunStr, _ := default_opts[hashOptionName].(string)
+				hashFunCode, ok := mh.Names[strings.ToLower(hashFunStr)]
+
+				if !ok {
+					return fmt.Errorf("unrecognized hash function: %s", strings.ToLower(hashFunStr))
+				}
+				my_opts := []options.UnixfsAddOption{
+					options.Unixfs.Hash(hashFunCode),
+					options.Unixfs.Inline(inline),
+					options.Unixfs.InlineLimit(inlineLimit),
+					options.Unixfs.Chunker(chunker),
+
+					options.Unixfs.Pin(dopin),
+					options.Unixfs.HashOnly(true),
+					options.Unixfs.FsCache(fscache),
+					options.Unixfs.Nocopy(nocopy),
+
+					options.Unixfs.Progress(progress),
+					options.Unixfs.Silent(silent),
+				}
+
+				if cidVerSet {
+					my_opts = append(my_opts, options.Unixfs.CidVersion(cidVer))
+				}
+
+				if rbset {
+					my_opts = append(my_opts, options.Unixfs.RawLeaves(rawblks))
+				}
+
+				if trickle {
+					my_opts = append(my_opts, options.Unixfs.Layout(options.TrickleLayout))
+				}
+
+				my_opts = append(my_opts, nil) // events option placeholder
+
+				fmt.Println(my_opts)
+
+				// var added int
+				addit := special.Entries()
+
+				for addit.Next() {
```

Comment:
`for addit.Next()` doesn't initiate a loop on the second default settings. Seems like the entries are removed on each loop. Is there a way we can persist the file as we try different params?

Reply:
Technically, you could buffer them all into memory. But that's _really_ not something you want to do.